### ISSUE TITLE
baseApiUrl w/ path & normalizeTileURL functionality

### ISSUE THREAD

**Issue by sumarlidason** [mapbox/mapbox-gl-js#8458]
**mapbox-gl-js version**: v0.54 / 1.0.0 / master

**browser**: chrome

I am experiencing malformed URLs generated from `normalizeTileURL` and `makeAPIURL`. This issue seems to only occur when the baseAPIURL contains a path (as opposed to just a host). I want to compare the behavior between mapbox.com & an Atlas hosted solution w/ a path of `https://localhost:8080/mbx`.

### .com
```
https://api.mapbox.com/v4/mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7.json
```
is returning this tile format string:
```
https://a.tiles.mapbox.com/v4/mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7/{z}/{x}/{y}.vector.pbf
```
It enters the `normalizeTileURL` function as:
```
mapbox://tiles/mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7/10/184/401.vector.pbf
```
manipulated within the function to
```
/v4/mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7/10/184/401.vector.pbf
```
and finally, from `makeAPIURL`:
```
https://api.mapbox.com/v4/mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7/10/184/401.vector.pbf
```

### Atlas Hosted
```
http://localhost:8080/mbx/v4/mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7.json
```
is returning me a semi-accurate tile format string (the protocol is wrong):
```
https://localhost:8080/mbx/v4/mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7/{z}/{x}/{y}.vector.pbf
```
It enters the `normalizeTileURL` function this way, however inside this function the path is manipulated, and the `v4` prefix is needlessly appended:
```
/v4/mbx/v4/mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7/10/184/401.vector.pbf
```
In the return statement this urlObject is passed to `makeAPIURL`, for a final url of
```
"http://localhost:8080/mbx/v4/mbx/v4/mapbox.mapbox-terrain-v2,mapbox.mapbox-streets-v7/10/184/401.vector.pbf"
```
You can see we got the API path prepended again.

## Seeking Guidance
I don't understand the rewriting of the url format provided by the `/v4` request. Additionally it looks like we are losing the subdomaining browser hack for increased parallelized requests?

**Comment by sumarlidason** [comment]
## Executing Test Case
```bash
git fetch
git checkout 8458_debugNormalizeTileURL
yarn test-unit util/mapbox.test.js
```

This test replicates the issue I am seeing above:
![image](https://user-images.githubusercontent.com/1991770/61013895-b91f6900-a352-11e9-8e29-a0b0a36c8fc1.png)

**Comment by sumarlidason** [comment]
Possible fix w/ 37ab6b7

1. added regex to trim the tileURL:
https://github.com/mapbox/mapbox-gl-js/blob/37ab6b7dc0247e3f88ae194ab072ad8f82bd6983/src/util/mapbox.js#L174-L195
2. made makeAPIURL a bit smarter about params, removing any existing access_token that was provided and replacing with the mapbox-gl-js configured token.
https://github.com/mapbox/mapbox-gl-js/blob/37ab6b7dc0247e3f88ae194ab072ad8f82bd6983/src/util/mapbox.js#L121-L124

I am not certain this is the best solution -
